UNKNOWN:Error received from peer ipv4:10.36.78.168:50151 {grpc_message:"Socket closed", grpc_status:14,

grpc connection gets terminated and after that cannot be restarted
i have to stop the container and start it again
my assumption is that `opi-vici-bridge` is crashing

attached is the swanctl.conf that I use and demo works
the python file is the OPI api I try to use on top of a swanctl empty config trying to achieve same result
[ipsec-config.zip](https://github.com/opiproject/opi-strongswan-bridge/files/10531409/ipsec-config.zip)

Nvidia Blue Field2
install DOCA_1.5.1_BSP_3.9.3_Ubuntu_20.04-4.2211-LTS.signed.bfb
```
docker pull ghcr.io/opiproject/opi-security-server:main

sudo sed -i "s/# socket = unix:\/\/\${piddir}\/charon\.vici/socket = unix:\/\/\/var\/run\/charon\.vici/g" /etc/strongswan.d/charon/vici.conf

docker run --rm --network host \
    --mount src=/var/run,target=/var/run,type=bind \
    -d ghcr.io/opiproject/opi-security-server:main /opi-vici-bridge -port=50151
```

on another machine get `https://github.com/opiproject/opi-api/blob/main/security/v1/gen/python`
```
root@test:~# python3
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import grpc
>>> import ipsec_pb2
>>> import ipsec_pb2_grpc
>>>
>>> channel = grpc.insecure_channel('10.36.78.168:50151')
>>> stub = ipsec_pb2_grpc.IPsecStub(channel)
>>>
>>> stub.IPsecVersion(ipsec_pb2.IPsecVersionReq())
daemon: "charon"
version: "5.9.6bf"
sysname: "Linux"
release: "5.4.0-1049-bluefield"
machine: "aarch64"

>>>
>>> stub.IPsecStats(ipsec_pb2.IPsecStatsReq())
status: "Running time: 5 hours\nAbsolute startup time: Jan 29 02:45:22 2023\nTotal # of worker threads: 16\nWorker threads currently idle: 11\nThreads processing critical priority jobs: 4\nThreads processing high priority jobs: 0\nThreads processing medium priority jobs: 1\nThreads processing low priority jobs: 0\nJobs queued with critical priority: 0\nJobs queued with high priority: 0\nJobs queued with medium priority: 0\nJobs queued with low priority: 0\n# of jobs scheduled for timed execution: 0\nTotal number of IKE_SAs active: 0\nNumber of IKE_SAs in half-open state: 0\nPlugins: charon aes des rc2 sha2 sha1 md5 mgf1 nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs12 pgp dnskey sshkey pem openssl pkcs8 fips-prf gmp curve25519 xcbc cmac hmac kdf drbg attr kernel-netlink resolve socket-default stroke vici updown xauth-generic counters \nNon-mmap\'d space available: 2703360\nMmap\'d space available: 0\nTotal number of bytes used: 547648\nAvailable but unsued bytes: 2155712\n"

>>>
>>>
>>>
>>>
                crypto_alg=[ipsec_pb2.AES128],
                integ_alg=[ipsec_pb2.SHA1]
            ),
            remote_ts=ipsec_pb2.TrafficSelectors(
                ts=[ipsec_pb2.TrafficSelectors.TrafficSelector(
                    cidr='40.0.1.0/24'
                )]
            ),
            local_ts=ipsec_pb2.TrafficSelectors(
                ts=[ipsec_pb2.TrafficSelectors.TrafficSelector(
                    cidr='201.0.0.0/24'
                )]
            ),
        )],
        proposals=ipsec_pb2.Proposals(
            crypto_alg=[ipsec_pb2.AES128],
            integ_alg=[ipsec_pb2.SHA384],
            dhgroups=[ipsec_pb2.MODP1536]
        )
    )
)>>> tun1_0_0 = ipsec_pb2.IPsecLoadConnReq(
...     connection=ipsec_pb2.Connection(
...         name='tun1_0_0',
...         version='2',
...         local_addrs=[ipsec_pb2.Addrs(addr='200.0.0.1')],
...         remote_addrs=[ipsec_pb2.Addrs(addr='200.0.0.2')],
...         local_auth=ipsec_pb2.LocalAuth(
...             auth=ipsec_pb2.PSK,
...             id='200.0.0.1'
...         ),
...         remote_auth=ipsec_pb2.RemoteAuth(
...             auth=ipsec_pb2.PSK,
...             id='200.0.0.2'
...         ),
...         children=[ipsec_pb2.Child(
...             name='tun1_0_0',
...             esp_proposals=ipsec_pb2.Proposals(
...                 crypto_alg=[ipsec_pb2.AES128],
...                 integ_alg=[ipsec_pb2.SHA1]
...             ),
...             remote_ts=ipsec_pb2.TrafficSelectors(
...                 ts=[ipsec_pb2.TrafficSelectors.TrafficSelector(
...                     cidr='40.0.1.0/24'
...                 )]
...             ),
...             local_ts=ipsec_pb2.TrafficSelectors(
...                 ts=[ipsec_pb2.TrafficSelectors.TrafficSelector(
...                     cidr='201.0.0.0/24'
...                 )]
...             ),
...         )],
...         proposals=ipsec_pb2.Proposals(
...             crypto_alg=[ipsec_pb2.AES128],
...             integ_alg=[ipsec_pb2.SHA384],
...             dhgroups=[ipsec_pb2.MODP1536]
...         )
...     )
... )
>>>
>>>
>>>
>>>
>>> stub.IPsecLoadConn(tun1_0_0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.10/dist-packages/grpc/_channel.py", line 946, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "/usr/local/lib/python3.10/dist-packages/grpc/_channel.py", line 849, in _end_unary_response_blocking
    raise _InactiveRpcError(state)
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.UNAVAILABLE
        details = "Socket closed"
        debug_error_string = "UNKNOWN:Error received from peer ipv4:10.36.78.168:50151 {grpc_message:"Socket closed", grpc_status:14, created_time:"2023-01-29T08:32:32.329483978+00:00"}"
>
>>>


```

i ran the `opi-security-server` in interactive mode and I got the corresponding failure from the server side

```
2023/01/29 20:43:54 IPsecLoadConn: Received: name:"tun1_0_0" version:"2" local_addrs:{addr:"200.0.0.1"} remote_addrs:{addr:"200.0.0.2"} proposals:{crypto_al:AES128 integ_alg:SHA384 dhgroups:MODP1536} local_auth:{auth:PSK id:"200.0.0.1"} remote_auth:{auth:PSK id:"200.0.0.2"} children:{name:"tun1_0_0" esp_proposas:{crypto_alg:AES128 integ_alg:SHA1} local_ts:{ts:{cidr:"201.0.0.0/24"}} remote_ts:{ts:{cidr:"40.0.1.0/24"}}}
2023/01/29 20:43:54 Local Port [500] Remote Port [500]
2023/01/29 20:43:54 DUMPING conn.Local: {psk [] 200.0.0.1    []}
2023/01/29 20:43:54 DUMPING conn.Remove: {psk 200.0.0.2  [] [] [] [] []}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x3b174c]

goroutine 10 [running]:
main.loadConn(0x40001adda0)
        /app/ipsec_vici.go:532 +0x70c
main.(*server).IPsecLoadConn(0x4516c0?, {0x40001adda0?, 0x0?}, 0x40001adda0)
        /app/ipsec.go:120 +0x64
github.com/opiproject/opi-api/security/v1/gen/go._IPsec_IPsecLoadConn_Handler({0x47b160?, 0x886b78}, {0x5690e0, 0x40001add70}, 0x400017be30, 0x0)
        /go/pkg/mod/github.com/opiproject/opi-api@v0.0.0-20230126195639-3f4a4c1386bc/security/v1/gen/go/ipsec.pb.go:5588 +0x16c
google.golang.org/grpc.(*Server).processUnaryRPC(0x40001a01e0, {0x56bb58, 0x40001aa820}, 0x40003427e0, 0x40001acb70, 0x84ac60, 0x0)
        /go/pkg/mod/google.golang.org/grpc@v1.52.3/server.go:1336 +0xb68
google.golang.org/grpc.(*Server).handleStream(0x40001a01e0, {0x56bb58, 0x40001aa820}, 0x40003427e0, 0x0)
        /go/pkg/mod/google.golang.org/grpc@v1.52.3/server.go:1704 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /go/pkg/mod/google.golang.org/grpc@v1.52.3/server.go:965 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /go/pkg/mod/google.golang.org/grpc@v1.52.3/server.go:963 +0x294

```

--- a/server/ipsec_vici.go
+++ b/server/ipsec_vici.go
@@ -529,9 +529,12 @@ func loadConn(connreq *pb.IPsecLoadConnReq) error {
 		log.Printf("DUMPING conn.Remove: %v", conn.Remote)
 	}
 
-	for i := 0; i < len(c.Vips.Vip); i++ {
-		conn.Vips = append(conn.Vips, c.Vips.Vip[i])
+	if c.Vips != nil {
+		for i := 0; i < len(c.Vips.Vip); i++ {
+			conn.Vips = append(conn.Vips, c.Vips.Vip[i])
+		}
 	}
+
 	for i := 0; i < len(c.LocalAddrs); i++ {
 		conn.LocalAddrs = append(conn.LocalAddrs, c.LocalAddrs[i].GetAddr())
 	}

